lib/github-checks.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const chalk = require('chalk');
+const execa = require('execa');
+const ghGot = require('gh-got');
+const hostedGitInfo = require('hosted-git-info');
+const pTimeout = require('p-timeout');
+const pWaitFor = require('p-wait-for');
+const terminalLink = require('terminal-link');
+const util = require('./util');
+
+const THIRTY_SECONDS = 1000 * 30;
+const THIRTY_MINUTES = 1000 * 60 * 30;
+
+module.exports = async task => {
+	const hasUnpushedCommits = await execa.stdout('git', ['cherry']) !== '';
+
+	if (hasUnpushedCommits) {
+		await execa('git', ['push']);
+	}
+
+	const latestCommit = execa.stdout('git', ['rev-parse', 'HEAD']);
+	const {title} = task;
+	const pkg = util.readPkg();
+	const {user, project} = hostedGitInfo.fromUrl(pkg.repository.url);
+
+	const checkState = async () => {
+		try {
+			const response = await ghGot(`repos/${user}/${project}/commits/${latestCommit}/status`);
+			const {state, statuses} = response.body;
+
+			if (state === 'failure') {
+				task.title = title;
+
+				const failures = statuses.filter(status => status.state === 'error' || status.state === 'failure');
+				const contexts = failures.map(status => terminalLink(status.context, status.target_url));
+
+				if (failures.length === statuses.length) {
+					throw new Error(`All checks have failed: ${contexts.join(', ')}`);
+				} else {
+					throw new Error(`Some checks were not successful: ${contexts.join(', ')}`);
+				}
+			} else if (state === 'pending') {
+				task.title = `${title} ${chalk.yellow('(waiting for pending checks…)')}`;
+
+				return false;
+			}
+
+			task.title = title;
+
+			return true;
+		} catch (error) {
+			const {rateLimit} = error;
+
+			if (rateLimit.remaining === 0) {
+				throw new Error('Exceeded API rate limit');
+			}
```

Comment:
Do we actually need this? It would already throw an error with the message `API rate limit exceeded for xxx.xxx.xxx.xxx. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)`.

https://developer.github.com/v3/#rate-limiting

Reply:
I figured a more succinct error message would be nicer. Do you think just throwing the HTTP error is enough?